tasks/kernel_matrix_testing/compiler.py

```diff
@@ -76,11 +76,15 @@ def start(self) -> None:
         if self.is_running:
             self.stop()
 
-        info("[+] Updating build image, ensuring we can log in to the registry")
-        self.ctx.run(
-            "aws-vault exec sso-build-stable-developer -- aws ecr --region us-east-1 get-login-password | docker login --username AWS --password-stdin 486234852809.dkr.ecr.us-east-1.amazonaws.com"
-        )
-        self.ctx.run(f"docker pull {self.image}")
+        # docker images command does not fail if the image is not found, so we need to grep the output
+        # to get an error exit code
+        res = self.ctx.run(f"docker images {self.image} | grep {self.image}", hide=True, warn=True)
```

Comment:
I think you can do `docker image inspect {self.image}` and just check the exit code.

Reply:
Done!